Templates/CDS-CSharp/readme.md

```diff
@@ -0,0 +1,458 @@
+        {
+            // Prompt for login and create security token
+            var authenticationContext = new AuthenticationContext(Authority);
+            AuthenticationResult authenticationResult = await authenticationContext.AcquireTokenAsync(FunctionResourceId
+                , ClientAppId
+                , new Uri(RedirectUri)
+                , new PlatformParameters(PromptBehavior.Always));
+            var securityTokenString = authenticationResult.CreateAuthorizationHeader();
+
+            // Create and configure the HTTP client
+            var client = new HttpClient();
+            var authorizationHeaderParameter = securityTokenString.Replace(AuthorizationHeaderScheme + " ", string.Empty);
+            client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue(AuthorizationHeaderScheme, authorizationHeaderParameter);
+            return client;
+        }
+```
+
+Configure the target environment and security setting of the app by replacing the corresponding bracket text in code with configuration values:
+
+1. **AAD tenant** should replace `[[Replace with AAD tenant value]]`.
+1. **AAD client application ID** should replace `[[Replace with AAD client application ID value]]`.
+1. **AAD client application redirect URI** should replace `[[Replace with AAD client redirect URI value]]`.
+1. **Function URL** should replace `[[Replace with Function URL value]]`.
+1. **AAD function resource ID** should replace `[[Replace with AAD function resource ID value]]`. **Note** that this **Resource ID** value comes from the **Function application** not the **custom API application**, and will look like the following example: `https://[your_domain_name]/4598e084-55a7-4d57-88ae-59902d8e3abc`.
+
+### Compile and run the project
+
+1. Ensure the project compiles by right clicking on the project and clicking **Build**.
+1. Run the client code by clicking on **Start** or pressing **F5**.
+1. Login using **your credentials** when the Azure AD prompt appears. The first time you run the application, you will be prompted to allow the AAD application you registered earlier to access the services Common Data Service uses.
+1. Verify that the program runs and calls the Function.
+1. Verify that the function updates Product Categories as expected.
+1. **Note** that if you see log errors related to assembly loading please refer to **Assembly load issue after publish to Azure** in the troubleshooting section. 
+
+# Advanced - PowerApps Custom API creation and configuration
+In order to call the Azure Function from an app, it first needs to be wrapped by a Custom API that defines its API structure and authentication settings.
+
+Prepare the [swagger](http://swagger.io/) file that defines your function by copying the content below to a .json file and modifying the **host** and **default code** values. You can get these values by going to the Azure Function edit page and clicking on **Copy function URL** on the top right.
```

Comment:
If you can just tell them how to modify the generated swagger from the functions page, it may be easier. Like remove all operations except GET and modify the the security definitions